src/lib/kafkajs/JSProducer.ts

```diff
@@ -94,25 +94,28 @@ export class JSProducer extends EventEmitter {
       throw new Error("You are missing a broker or group configs");
     }
 
+    const conf = {
+      brokers,
+      clientId,
+    } as KafkaConfig
     if (securityProtocol) {
-      this.kafkaClient = new Kafka({
-        brokers,
-        clientId,
-        ssl: {
+      if (securityProtocol.includes('sasl')) {
```

Comment:
This could be written better as a switch e.g
```
switch (securityProtocol) {
  case 'sasl':
    conf.sasl = {
      mechanism: mechanism as SASLMechanism,
      username: username as string,
      password: password as string,
    }
    break;
 case 'ssl':
   conf.ssl = {
     ca: [fs.readFileSync(sslCALocation as string, "utf-8")],
     cert: fs.readFileSync(sslCertLocation as string, "utf-8"),
     key: fs.readFileSync(sslKeyLocation as string, "utf-8"),
     passphrase: sslKeyPassword,
  }
  break;
}
```

Reply:
ah, so is `sasl_ssl` no longer supported? happy to go either way, just not sure which should be pursued

Reply:
Ah, sorry no it is. My mistake.